http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtHttpClient.java

```diff
@@ -0,0 +1,286 @@
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.crt;
+
+import static software.amazon.awssdk.http.HttpMetric.HTTP_CLIENT_NAME;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Consumer;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.crt.http.HttpClientConnectionManager;
+import software.amazon.awssdk.crt.http.HttpException;
+import software.amazon.awssdk.http.ExecutableHttpRequest;
+import software.amazon.awssdk.http.HttpExecuteRequest;
+import software.amazon.awssdk.http.HttpExecuteResponse;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.SdkHttpConfigurationOption;
+import software.amazon.awssdk.http.SdkHttpFullResponse;
+import software.amazon.awssdk.http.crt.internal.AwsCrtClientBuilderBase;
+import software.amazon.awssdk.http.crt.internal.AwsCrtHttpClientBase;
+import software.amazon.awssdk.http.crt.internal.CrtRequestContext;
+import software.amazon.awssdk.http.crt.internal.CrtRequestExecutor;
+import software.amazon.awssdk.metrics.NoOpMetricCollector;
+import software.amazon.awssdk.utils.AttributeMap;
+
+/**
+ * An implementation of {@link SdkHttpClient} that uses the AWS Common Runtime (CRT) Http Client to communicate with
+ * Http Web Services. This client has a synchronous interface, but uses non-blocking IO.
+ *
+ * <p>This can be created via {@link #builder()}</p>
+ * {@snippet :
+ * SdkHttpClient client = software.amazon.awssdk.http.crt.AwsCrtHttpClient.builder()
+ * .maxConcurrency(100)
+ * .connectionTimeout(Duration.ofSeconds(1))
+ * .connectionMaxIdleTime(Duration.ofSeconds(5))
+ * .build();
+ *}
+ *
+ */
+@SdkPublicApi
+public final class AwsCrtHttpClient extends AwsCrtHttpClientBase implements SdkHttpClient {
+
+    private AwsCrtHttpClient(DefaultBuilder builder, AttributeMap config) {
+        super(builder, config);
+    }
+
+    public static AwsCrtHttpClient.Builder builder() {
+        return new DefaultBuilder();
+    }
+
+    /**
+     * Create a {@link AwsCrtHttpClient} client with the default configuration
+     *
+     * @return an {@link SdkHttpClient}
+     */
+    public static AwsCrtHttpClient create() {
+        return new DefaultBuilder().build();
+    }
+
+    @Override
+    public String clientName() {
+        return SdkHttpClient.super.clientName();
+    }
+
+    @Override
+    public ExecutableHttpRequest prepareRequest(HttpExecuteRequest request) {
+        request.metricCollector()
+               .filter(metricCollector -> !(metricCollector instanceof NoOpMetricCollector))
```

Comment:
Honestly I think we should just remove this filter. If it's the no-op collector it the `reportMetric` should have no effect

Reply:
I have removed the filter from both async and sync client.